src/qt/askpassphrasedialog.cpp

```diff
@@ -264,21 +264,25 @@ void AskPassphraseDialog::textChanged()
 
 bool AskPassphraseDialog::event(QEvent* event)
 {
-    // Detect Caps Lock key press.
     if (event->type() == QEvent::KeyPress) {
         QKeyEvent* ke = static_cast<QKeyEvent*>(event);
+        // Detect Caps Lock key press.
         if (ke->key() == Qt::Key_CapsLock) {
             fCapsLock = !fCapsLock;
+            if (fCapsLock) {
+                ui->capsLabel->setText(tr("Warning: The Caps Lock key is on!"));
+                ui->capsLabel->setVisible(true);
+            } else {
+                ui->capsLabel->clear();
+                ui->capsLabel->setVisible(false);
+            }
```

Comment:
This needs to be checked after **any** key press 
(i.e. `if (fCapsLock)` needs to be outside of `if (ke->key() == Qt::Key_CapsLock)` as it was).

Reply:
Why?

Reply:
On a second thought, it's probably fine. But why move it in the first place?

Reply:
I think it makes more sense to have one block per checked key handling all actions related to that key. Rather than having checks specifics to a key out of the block of this key.

Reply:
agree